Library/brew.rb

```diff
@@ -39,13 +38,24 @@
   EOS
 end
 
+# Check for user agreement of the Xcode license before permitting
+# any other brew usage to continue. This prevents the situation where
+# people are instructed to "please re-run as root via sudo" on brew commands.
+# The check can only fail when Xcode is installed & the active developer dir.
+if `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
+  odie <<-EOS.undent
+    You have not agreed to the Xcode license. Please resolve this by running:
+      sudo xcodebuild -license
+  EOS
+end
```

Comment:
Aye. `if OS.mac? && MacOS::Xcode.installed?` was the original check there; It was just 0.03 seconds slower so I left it out initially. Didn't occur to me calling `clang` with no CLT/Xcode installed would throw the CLT dialogue, despite having seen it a thousand times :smile_cat: